src/Core/MySQL/MySQLUtils.h

```diff
@@ -0,0 +1,20 @@
+#include "Core/DecimalFunctions.h"
+#include "DataTypes/IDataType.h"
+#include "base/types.h"
+
+namespace DB
+{
+namespace MySQLProtocol
+{
+namespace MySQLUtils
+{
+/// Splits DateTime64 column data at a certain row number into whole and fractional part
+/// Additionally, normalizes the fractional part as if it was scale 6 for MySQL compatibility purposes
+DecimalUtils::DecimalComponents<DateTime64> getNormalizedDateTime64Components(DataTypePtr data_type, ColumnPtr col, int row_num);
+
+/// If a column is ColumnSparse/ColumnLowCardinality/ColumnNullable, it is unwrapped in a correct order;
+/// otherwise, the original column is returned
+ColumnPtr getBaseColumn(const DB::Columns & columns, size_t i);
```

Comment:
Maybe makes sense to move elsewhere if there is a use for it in the other parts of the code.

Reply:
Fine to keep it here, imho - the code is quite simple and easy to replicate.